sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ReadFromKafkaDoFn.java

```diff
@@ -343,6 +343,9 @@ public ProcessContinuation processElement(
     // Stop processing current TopicPartition when it's time to stop.
     if (checkStopReadingFn != null
         && checkStopReadingFn.apply(kafkaSourceDescriptor.getTopicPartition())) {
+      // Attempt to claim the last element in the restriction, such that the restriction tracker
+      // doesn't throw an exception when checkDone is called
```

Comment:
Is this documented / designed behavior of restriction tracker (tracker is considered done when the last element has been claimed, not all elements are claimed)?
Thinking about this because recently I was implementing something that turned out to rely on some internal behavior but then it broke as master evolves

Reply:
It isn't documented as such, but because offset range tracker is monotonically increasing, claiming the last offset means that all other offsets can't be claimed